bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableConnection.java

```diff
@@ -68,6 +72,39 @@
       Collections.synchronizedMap(new HashMap<Long, BigtableBufferedMutator>());
 
   static {
+    // Initialize some core dependencies in parallel.  This can speed up startup by 150+ ms.
+    ExecutorService connectionStartupExecutor =
+        Executors.newCachedThreadPool(
+            new ThreadFactoryBuilder()
+                .setNameFormat("AbstractBigtableConnection-startup-%s")
+                .setDaemon(true)
+                .build());
+    connectionStartupExecutor.execute(new Runnable() {
+      @Override
+      public void run() {
+        // Startup of SSL_CONTEXT_FACTORY is a really expensive operation for the first operation.
+        // It could take up to 2 seconds.
+        BigtableOptions.SSL_CONTEXT_FACTORY.create();
+      }
+    });
+    connectionStartupExecutor.execute(new Runnable() {
+      @Override
+      public void run() {
+        // Initializing this variable is expensive. It should happen once at startup asynchronously.
+        @SuppressWarnings("unused")
+        Object ignored = BigtableServiceGrpc.CONFIG;
```

Comment:
Use BigtableServiceServiceDescriptor for clarity? How come this happens when assigning the variable? would it be worth asking grpc team to initialize it at startup?